lib/trip_dispatcher.rb

```diff
@@ -27,15 +34,50 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def request_trip(passenger_id)
+      passenger = find_passenger(passenger_id)
+      if !passenger
+        return nil
+      end
+
+      avail_driver = nil
+      @drivers.each do |driver|
+        if driver.status == :AVAILABLE
+          avail_driver = driver
+          break
```

Comment:
instead of using a `break` here, consider whether there's an enumerable method or whether a `while` loop would work.